### PR TITLE
More bona fide

### LINKED ISSUE #3000: Make Resources block hideable
Make Resources block hideable by config. For example bona fide has only one item (ever).

### LINKED ISSUE #3001: Make PDF (action) button hideable
Make PDF (action) button hideable by config. For example bona fide doesn't need it.

### LINKED ISSUE #3079: Try later versions of `react-select`.
We should upgrade from `react-select` v2 to v5 or so.

### LINKED ISSUE #3080: Fix guide GitHub links
Links from guide to GitHub don't work.

### PR DESCRIPTION
**Pull request by Macroz**
Close #3000 
Close #3001 
Close #3079
Close #3080

Implements all the missing features for #2836. The idea is to make the experience as simplified as possible.

- The location of the success message of a submit has been changed from the top of the page to the Actions block. (#2836)
  This is more consistent with the other actions that provide feedback in the same location. (#2836)
- The applicaiton Applicants block has a simple version for the case of a single applicant. (#2836)
- The Decide action button has been changed to primary action, as it usually is for a person in the `:decider` role. (#2836)
- Visibility of the PDF and ZIP download actions can now be configured with `:show-pdf-action` and `:show-attachment-zip-action` (#3001)
- The Resources block can be hidden from the application view. This can be useful if there are few catalogue items. (#3000)
- The application title field, in the application state block is now shown only if the application description is not blank. (#2836)- The previous applications block is now shown only if there are some. (#2836)
- The application Actions block is now successfully hidden, if there are no actions available (#2836)
- The component guide page had accumulated some small errors that are now fixed. (#2836)

# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [x] Link to issue
- [x] Consider adding screenshots for ease of review

## Backwards compatibility
- [x] Config is backwards compatible

## Documentation
- [x] Update changelog if necessary
- [x] Components are added to guide page
- [ ] Update docs/ (if applicable)
- [x] New config options in config-defaults.edn

## Testing
- [ ] Complex logic is unit tested
- [ ] Valuable features are integration / browser / acceptance tested automatically

## Follow-up
- [ ] New tasks are created for pending or remaining tasks

bona-fide1
bona-fide2

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
     "puppeteer": "^14.1.2",
-    "shadow-cljs": "^2.16.6"
-  },
-  "scripts": {
-    "dev": "shadow-cljs watch app"
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "shadow-cljs": "^2.16.6",
+    "diff-match-patch": "^1.0.5",
+    "react-select": "^2.4.4"
```
Comment (aatkin) [inline package.json:12]: updating `react@^17.0.2` seems to cause peer-dependency issue with `react-select@^2.4.4` when running `npm install`:
```sh
...
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^15.3.0 || ^16.0.0" from react-select@2.4.4
npm ERR! node_modules/react-select
npm ERR!   dev react-select@"^2.4.4" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: react@16.14.0
npm ERR! node_modules/react
npm ERR!   peer react@"^15.3.0 || ^16.0.0" from react-select@2.4.4
npm ERR!   node_modules/react-select
npm ERR!     dev react-select@"^2.4.4" from the root project
...
```

i think we need to look at updating `react-select` to a newer version, but there are quite a few majors between ours and latest.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,7 +8,16 @@ have notable changes.
 
 Changes since v2.30
 
+### Changes
+- The location of the success message of a submit has been changed from the top of the page to the Actions block. (#2836)
+  This is more consistent with the other actions that provide feedback in the same location. (#2836)
+- The applicaiton Applicants block has a simple version for the case of a single applicant. (#2836)
```
Comment (aatkin) [inline CHANGELOG.md:14]: tiny typo `The applicaiton` -> `The application`

**Review by aatkin — APPROVED** [review]
lots of good fixes and features! :+1: check package.json version issue with react, otherwise looks great.

File: package.json (hunk 1 above)
Comment (Macroz) [reply package.json:12]: Yes, I added an idea about this. Not sure if it's a problem since everything seems to work ...

Comment (Macroz) [reply package.json:12]: Ok, seems like the upgrade was easy. Let's test this manually in demo. Issue in #3079.

**Pull request merged**